contracts/run_tests.sh

```
@@ -0,0 +1,137 @@
+FAILED_TESTS=0
+
+# Function to run test suite
+run_test_suite() {
+    local suite_name=$1
+    local test_pattern=$2
+    
+    echo -e "${BLUE}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${NC}"
+    echo -e "${BLUE}Running: ${suite_name}${NC}"
+    echo -e "${BLUE}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${NC}"
+    
+    if cargo test ${test_pattern} -- --nocapture --test-threads=1 2>&1 | tee test_output.log; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)